dotnet/Razorvine.Pyrolite/Pyrolite/Pickle/Unpickler.cs

```diff
@@ -29,11 +33,26 @@ public class Unpickler : IDisposable {
 	protected readonly IDictionary<int, object> memo;
 	protected UnpickleStack stack;
 	protected Stream input;
-	protected static readonly IDictionary<string, IObjectConstructor> objectConstructors;
+	protected static readonly IDictionary<string, IObjectConstructor> objectConstructors = CreateObjectConstructorsDictionary();
 	protected static readonly object NO_RETURN_VALUE = new object();
-
-	static Unpickler() {
-		objectConstructors = new Dictionary<string, IObjectConstructor>
+    private static readonly string[] quoteStrings = new [] { "\"", "'" };
+    private static readonly object boxedFalse = false;
+    private static readonly object boxedTrue = true;
+    private byte[] byteBuffer = new byte[sizeof(long)]; // at least large enough for any primitive being deserialized
+    private Dictionary<StringPair, string> concatenatedModuleNames;
+
+    private readonly struct StringPair : IEquatable<StringPair>
+    {
+        public readonly string Item1, Item2;
+        public StringPair(string item1, string item2) { Item1 = item1; Item2 = item2; }
+        public bool Equals(StringPair other) => Item1 == other.Item1 && Item2 == other.Item2;
```

Comment:
Do we need to check `other` for `null` here?

Reply:
No, StringPair is a struct.